hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieNativeAvroHFileReader.java

```diff
@@ -244,14 +248,32 @@ private static GenericRecord getRecordFromKeyValue(KeyValue keyValue,
         readerSchema);
   }
 
-  private synchronized HFileReader getSharedHFileReader() {
-    try {
-      if (!sharedHFileReader.isPresent()) {
-        sharedHFileReader = Option.of(newHFileReader());
+  private byte[] getHFileMetaInfoFromCache(String key) throws IOException {
```

Comment:
I think we also want to put synchronized on this method: https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html.

Reply:
The new logic always checks the `isMetaInfoLoaded` before getting the value.